### ISSUE TITLE
[BUG] No such middleware to insert after: ActionDispatch::DebugExceptions

### ISSUE THREAD

**Issue by b0rdjack** [jnunemaker/flipper#623]
We wanted to add the `flipper` and `flipper-active_record` gem to our app (Rails 6), but once we run `rails g flipper:active_record` we get this error:

```
/home/user/.rvm/gems/ruby-2.6.9/gems/actionpack-6.0.4.7/lib/action_dispatch/middleware/stack.rb:144:in `assert_index': No such middleware to insert after: ActionDispatch::DebugExceptions (RuntimeError)
        from /home/user/.rvm/gems/ruby-2.6.9/gems/actionpack-6.0.4.7/lib/action_dispatch/middleware/stack.rb:109:in `insert_after'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/configuration.rb:51:in `block in insert_after'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/configuration.rb:76:in `block in merge_into'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/configuration.rb:75:in `each'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/configuration.rb:75:in `merge_into'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/engine.rb:512:in `block in app'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/engine.rb:509:in `synchronize'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/engine.rb:509:in `app'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/application/finisher.rb:97:in `block in <module:Finisher>'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/initializable.rb:32:in `instance_exec'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/initializable.rb:32:in `run'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/initializable.rb:61:in `block in run_initializers'
        from /home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:228:in `block in tsort_each'
        from /home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
        from /home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:431:in `each_strongly_connected_component_from'
        from /home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:349:in `block in each_strongly_connected_component'
        from /home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:347:in `each'
        from /home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:347:in `call'
        from /home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:347:in `each_strongly_connected_component'
        from /home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:226:in `tsort_each'
        from /home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:205:in `tsort_each'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/initializable.rb:60:in `run_initializers'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/application.rb:363:in `initialize!'
        from /home/user/development/projects/demo/config/environment.rb:19:in `<main>'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/zeitwerk-2.5.4/lib/zeitwerk/kernel.rb:35:in `require'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/activesupport-6.0.4.7/lib/active_support/dependencies.rb:324:in `block in require'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/activesupport-6.0.4.7/lib/active_support/dependencies.rb:291:in `load_dependency'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/activesupport-6.0.4.7/lib/active_support/dependencies.rb:324:in `require'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/application.rb:339:in `require_environment!'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/command/actions.rb:28:in `require_environment!'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/command/actions.rb:15:in `require_application_and_environment!'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/commands/generate/generate_command.rb:21:in `perform'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/thor-1.2.1/lib/thor/command.rb:27:in `run'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/thor-1.2.1/lib/thor/invocation.rb:127:in `invoke_command'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/thor-1.2.1/lib/thor.rb:392:in `dispatch'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/command/base.rb:69:in `perform'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/command.rb:46:in `invoke'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/commands.rb:18:in `<main>'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
        from /home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
        from bin/rails:6:in `<main>'
```

We also tried to run `rake middleware` and got this error:
```
No such middleware to insert after: ActionDispatch::DebugExceptions
/home/user/.rvm/gems/ruby-2.6.9/gems/actionpack-6.0.4.7/lib/action_dispatch/middleware/stack.rb:144:in `assert_index'
/home/user/.rvm/gems/ruby-2.6.9/gems/actionpack-6.0.4.7/lib/action_dispatch/middleware/stack.rb:109:in `insert_after'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/configuration.rb:51:in `block in insert_after'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/configuration.rb:76:in `block in merge_into'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/configuration.rb:75:in `each'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/configuration.rb:75:in `merge_into'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/engine.rb:512:in `block in app'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/engine.rb:509:in `synchronize'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/engine.rb:509:in `app'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/application/finisher.rb:97:in `block in <module:Finisher>'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/initializable.rb:32:in `instance_exec'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/initializable.rb:32:in `run'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/initializable.rb:61:in `block in run_initializers'
/home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:228:in `block in tsort_each'
/home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
/home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:431:in `each_strongly_connected_component_from'
/home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:349:in `block in each_strongly_connected_component'
/home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:347:in `each'
/home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:347:in `call'
/home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:347:in `each_strongly_connected_component'
/home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:226:in `tsort_each'
/home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/tsort.rb:205:in `tsort_each'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/initializable.rb:60:in `run_initializers'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/application.rb:363:in `initialize!'
/home/user/development/projects/demo/config/environment.rb:19:in `<main>'
/home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
/home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
/home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
/home/user/.rvm/gems/ruby-2.6.9/gems/bootsnap-1.7.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
/home/user/.rvm/gems/ruby-2.6.9/gems/zeitwerk-2.5.4/lib/zeitwerk/kernel.rb:35:in `require'
/home/user/.rvm/gems/ruby-2.6.9/gems/activesupport-6.0.4.7/lib/active_support/dependencies.rb:324:in `block in require'
/home/user/.rvm/gems/ruby-2.6.9/gems/activesupport-6.0.4.7/lib/active_support/dependencies.rb:291:in `load_dependency'
/home/user/.rvm/gems/ruby-2.6.9/gems/activesupport-6.0.4.7/lib/active_support/dependencies.rb:324:in `require'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/application.rb:339:in `require_environment!'
/home/user/.rvm/gems/ruby-2.6.9/gems/railties-6.0.4.7/lib/rails/application.rb:523:in `block in run_tasks_blocks'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/task.rb:281:in `block in execute'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/task.rb:281:in `each'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/task.rb:281:in `execute'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
/home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/monitor.rb:235:in `mon_synchronize'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/task.rb:243:in `block in invoke_prerequisites'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/task.rb:241:in `each'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/task.rb:241:in `invoke_prerequisites'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/task.rb:218:in `block in invoke_with_call_chain'
/home/user/.rvm/rubies/ruby-2.6.9/lib/ruby/2.6.0/monitor.rb:235:in `mon_synchronize'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/task.rb:188:in `invoke'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/application.rb:160:in `invoke_task'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/application.rb:116:in `each'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/application.rb:116:in `block in top_level'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/application.rb:125:in `run_with_threads'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/application.rb:110:in `top_level'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/application.rb:83:in `block in run'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/application.rb:186:in `standard_exception_handling'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/lib/rake/application.rb:80:in `run'
/home/user/.rvm/gems/ruby-2.6.9/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/home/user/.rvm/gems/ruby-2.6.9/bin/rake:23:in `load'
/home/user/.rvm/gems/ruby-2.6.9/bin/rake:23:in `<main>'
/home/user/.rvm/gems/ruby-2.6.9/bin/ruby_executable_hooks:22:in `eval'
/home/user/.rvm/gems/ruby-2.6.9/bin/ruby_executable_hooks:22:in `<main>'
```

We also downgraded the flipper version to the `0.23.0` and it works fine (migration runs successfully), but from the `0.23.2` to the latest version we get the error above, any clues ?

**Comment by jnunemaker** [comment]
Hi! What does rake middleware look like using 0.23?

**Comment by jnunemaker** [comment]
@b0rdjack I'm a little confused as to how flipper could have anything to do with this. I searched the entire repository and found no results for `DebugExceptions`. 

It does look like your middleware stack has `use QuietRoutesDebugExceptions` but googling returns nothing for that. Is that something custom that your app has? 

When you upgrade flipper to 0.24 do any other dependencies change in Gemfile.lock?

Closing to keep things tidy here but happy to still help debug what is going on. If we find a bug in flipper, I'll re-open.
